draft-tuexen-opsawg-pcapng.xml

```diff
@@ -1843,6 +1843,86 @@ Section Header
       </section>
 
 
+      <section anchor="section_dsb" title="Decryption Secrets Block">
+
+        <t>A Decryption Secrets Block (DSB) stores (session) secrets that
+        enable decryption of packets within the capture file. The format of
+        these secrets is defined by the Secrets Type.</t>
+
+        <t>Multiple DSBs can exist in a pcapng file, but they SHOULD be written
+        before packet blocks that require those secrets. Tools MAY limit
+        decryption to secrets that appear before packet blocks.</t>
+
+        <t>The structure of a
+        Decryption Secrets Block is shown in <xref target="format_dsb"/>.</t>
+
+        <figure anchor="format_dsb" title="Decryption Secrets Block Format">
+          <artwork xml:space="preserve" name="" type="" align="center" alt="" width="" height="">
+   0                   1                   2                   3
+   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +---------------------------------------------------------------+
+ 0 |                   Block Type = 0x0000000A                     |
+   +---------------------------------------------------------------+
+ 4 |                      Block Total Length                       |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ 8 |                          Secrets Type                         |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+12 |                         Secrets Length                        |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+16 /                                                               /
+   /                          Secrets Data                         /
+   /              (variable length, padded to 32 bits)             /
+   /                                                               /
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   /                                                               /
+   /                       Options (variable)                      /
+   /                                                               /
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   /                       Block Total Length                      /
+   +---------------------------------------------------------------+
+</artwork>
+        </figure>
+
+        <t>The Decryption Secrets Block has the following fields.
+          <list style="symbols">
+
+            <t>Block Type: The block type of the Decryption Secrets Block is
+            10.</t>
+
+            <t>Block Total Length: total size of this block, as described in
+            <xref target="section_block"/>.</t>
+
+            <t>Secrets Type: a 32-bit unsigned identifier that describes the
```

Comment:
Are these expected to be 4-character ASCII strings, like "TLSK", or do they just have to be unique?

(I'm going to update the spec to make it clearer that the same procedure should be used to request a standard block type and standard option codes for the block; this change should then say "...and to request standard secrets types for a DSB".)

Reply:
Did I pick an ASCII string that happens to be `TLSK`? What a coincidence! ;)

They are expected to be unique unsigned integers, I'll clarify this.